packages/react-router/modules/matchPath.js

```diff
@@ -30,7 +30,8 @@ const matchPath = (pathname, options = {}) => {
   if (typeof options === 'string')
     options = { path: options }
 
-  const { path, exact = false, strict = false } = options
+  const { exact = false, strict = false } = options
+  const path = options.path || options.from
```

Comment:
I think it's fine. `<Switch>` is kind of magical anyway. It reads the data of its `children` and decides what to do based on their props. We can pretty easily say `<Redirect from>` is only supported inside a `<Switch>`.